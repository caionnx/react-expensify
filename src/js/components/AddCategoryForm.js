import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import slugify from 'slugify'
import { startAddCategory } from '../actions/categories'

export class AddCategoryForm extends React.Component {
  state = {
    error: ''
  }

  onAddCategory = (event) => {
    event.preventDefault()
    const node = event.target.querySelector(`#new-category`)
    const value = node && node.value
    const id = slugify(value, { replacement: '-', remove: null, lower: true })

    if (id === '') {
      this.setState(() => ({ error: 'Please provide category name.' }))
      return
    }
    if (this.props.categories.filter((item) => item.id === id).length) {
      this.setState(() => ({ error: 'Category already registered.' }))
      return
    }

    this.setState(() => ({ error: '' }))
    this.props.startAddCategory({ id, value })
    node.value = ''
  }

  render () {
    const { error } = this.state

    return (
      <form onSubmit={this.onAddCategory}>
        {error && <p className='form__error'>{error}</p>}
        <div className='input-group'>
          <div className='input-group__item'>
            <input id='new-category' className='text-input' placeholder='New Category' type='text' />
          </div>
          <div className='input-group__item'>
            <button className='button'>Add Category</button>
          </div>
        </div>
      </form>
    )
  }
}

AddCategoryForm.propTypes = {
  categories: PropTypes.array,
  startAddCategory: PropTypes.func.isRequired
}

const mapStateToProps = ({ categories }) => ({
  categories
})

const mapDispatchToProps = (dispatch) => ({
  startAddCategory: (category) => dispatch(startAddCategory(category))
})

export default connect(mapStateToProps, mapDispatchToProps)(AddCategoryForm)
