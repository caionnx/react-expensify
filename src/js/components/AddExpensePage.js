import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import ExpenseForm from './ExpenseForm'
import { startAddExpense } from '../actions/expenses'

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense).then(() => {
      this.props.history.push('/')
    })
  }
  render () {
    return (
      <div>
        <div className='page-header'>
          <div className='content-container'>
            <h1 className='page-header__title'>Add Expense</h1>
          </div>
        </div>
        <div className='content-container'>
          <ExpenseForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    )
  }
}

AddExpensePage.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }),
  startAddExpense: PropTypes.func.isRequired
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
})

export default connect(undefined, mapDispatchToProps)(AddExpensePage)
