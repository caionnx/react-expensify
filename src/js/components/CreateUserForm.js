import React from 'react'
import { connect } from 'react-redux'
import { startCreateUser } from '../actions/auth'
import PropTypes from 'prop-types'

export class CreateUserForm extends React.Component {
  state = {
    error: false,
    loading: false
  }

  isEmail = (email) =>
    /(.+)@(.+){2,}\.(.+){2,}/.test(email)

  isPassword = (password) =>
    /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,8}$/.test(password)

  onFormSubmit = (ev) => {
    ev.preventDefault()
    const formData = new FormData(ev.target) // eslint-disable-line
    const email = formData.get('email')
    const confirmEmail = formData.get('confirm-email')
    const password = formData.get('password')

    if (!this.isEmail(email)) {
      this.setState(() => ({ error: 'Invalid email address.' }))
      return
    }

    if (email !== confirmEmail) {
      this.setState(() => ({ error: 'Please, confirm your email.' }))
      return
    }

    if (!this.isPassword(password)) {
      this.setState(() => ({ error: 'Password must be min 6 characters and max 8 characters. Must include one upper case letter, one lower case letter, and one numeric digit.' }))
      return
    }

    this.setState(() => ({ loading: true }))
    this.props.startCreateUser(email, password).catch((fail) => {
      const error = fail.code === 'auth/email-already-in-use'
        ? fail.message
        : 'Fail to create account.'

      this.setState(() => ({ error, loading: false }))
    })
  }

  render () {
    const { error, loading } = this.state
    const { goBackFunction } = this.props

    return (
      <div>
        <p>Create your account.</p>
        <form className='form' onSubmit={this.onFormSubmit}>
          <input placeholder='Email' aria-label='Email' type='text' name='email' required className='text-input' />
          <input placeholder='Confirm Email' aria-label='Confirm Email' type='text' name='confirm-email' required className='text-input' />
          <input placeholder='Password' aria-label='Password' type='password' name='password' required className='text-input' />
          { error && <p className='form__error'>{error}</p> }
          <button type='submit' className='button' disabled={loading}>
            { loading ? 'Submiting..' : 'Create' }
          </button>
          { goBackFunction &&
            <a href='#' className='button button--secondary' onClick={goBackFunction}>Cancel</a> }
        </form>
      </div>
    )
  }
}

CreateUserForm.propTypes = {
  startCreateUser: PropTypes.func,
  goBackFunction: PropTypes.func
}

const mapDispatchToProps = (dispatch) => ({
  startCreateUser: (emai, password) => dispatch(startCreateUser(emai, password))
})

export default connect(undefined, mapDispatchToProps)(CreateUserForm)
