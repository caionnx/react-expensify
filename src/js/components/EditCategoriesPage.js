import React from 'react'
import { connect } from 'react-redux'
import Modal from './Modal'
import slugify from 'slugify'
import PropTypes from 'prop-types'
import {
  startSetCategories,
  startRemoveCategory,
  startAddCategory
} from '../actions/categories'

export class EditCategoriesPage extends React.Component {
  state = {
    removeCategoryModal: false,
    idToRemove: false,
    error: ''
  }
  onOpenRemoveCategoryModal = (id) => {
    this.setState({ removeCategoryModal: true, idToRemove: id })
  }
  onCloseRemoveCategoryModal = () => {
    this.setState({ removeCategoryModal: false })
  }
  onRemove = () => {
    if (this.state.idToRemove) {
      this.props.startRemoveCategory({ id: this.state.idToRemove }).then(() => {
        this.setState({ removeCategoryModal: false })
      })
    }
  }
  onAddCategory = (event) => {
    event.preventDefault()
    const node = event.target.querySelector(`#new-category`)
    const value = node && node.value
    const id = slugify(value, { replacement: '-', remove: null, lower: true })

    if (id === '') {
      this.setState(() => ({ error: 'Please provide category name.' }))
    } else if (this.props.categories.filter((item) => item.id === id).length) {
      this.setState(() => ({ error: 'Category already registered.' }))
    } else {
      this.setState(() => ({ error: '' }))
      this.props.startAddCategory({ id, value })
      node.value = ''
    }
  }
  componentDidMount () {
    const { categories } = this.props

    if (categories && !categories.length) {
      this.props.fillCategories()
    }
  }

  render () {
    const { categories } = this.props

    return (
      <div>
        <div className='page-header'>
          <div className='content-container'>
            <h1 className='page-header__title'>Edit your categories</h1>
          </div>
        </div>
        <div className='content-container'>
          <div className='list-header'>
            <div className='show-for-mobile'>Categories</div>
            <div className='show-for-desktop'>Categories</div>
          </div>
          <div className='list-body'>
            {
              categories && categories.map(
                item =>
                  <div className='list-item' key={item.id}>
                    <h3 className='list-item__title'>{item.value}</h3>
                    <button onClick={() => this.onOpenRemoveCategoryModal(item.id)}>
                      <svg viewBox='0 0 12 12' width='12' height='12'>
                        <path fillRule='evenodd' d='M11.53.47a.75.75 0 0 0-1.061 0l-4.47 4.47L1.529.47A.75.75 0 1 0 .468 1.531l4.47 4.47-4.47 4.47a.75.75 0 1 0 1.061 1.061l4.47-4.47 4.47 4.47a.75.75 0 1 0 1.061-1.061l-4.47-4.47 4.47-4.47a.75.75 0 0 0 0-1.061z' />
                      </svg>
                    </button>
                  </div>
              )
            }
          </div>
          <Modal
            isOpen={this.state.removeCategoryModal}
            label='Remove category'
            className='modal'
            body='Are you sure you want to remove this category?'
            confirmText='Remove'
            cancelText='Cancel'
            thunk
            onConfirm={this.onRemove}
            onClose={this.onCloseRemoveCategoryModal} />

          {this.state.error && <p className='form__error'>{this.state.error}</p>}
          <form onSubmit={this.onAddCategory}>
            <div className='input-group'>
              <div className='input-group__item'>
                <input id='new-category' className='text-input' placeholder='New Category' type='text' />
              </div>
              <div className='input-group__item'>
                <button className='button'>Add Category</button>
              </div>
            </div>
          </form>
        </div>
      </div>
    )
  }
}

EditCategoriesPage.propTypes = {
  categories: PropTypes.array,
  fillCategories: PropTypes.func.isRequired,
  startRemoveCategory: PropTypes.func.isRequired,
  startAddCategory: PropTypes.func.isRequired
}

const mapStateToProps = ({ categories }) => ({
  categories
})

const mapDispatchToProps = (dispatch) => ({
  fillCategories: () => dispatch(startSetCategories()),
  startRemoveCategory: (uid) => dispatch(startRemoveCategory(uid)),
  startAddCategory: (category) => dispatch(startAddCategory(category))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditCategoriesPage)
