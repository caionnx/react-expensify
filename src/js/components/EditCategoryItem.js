import React from 'react'
import PropTypes from 'prop-types'

export const EditCategoryItem = ({ category, buttonAction }) => (
  <div className='list-item list-item--no-link' key={category.id}>
    <h3 className='list-item__title'>{category.value}</h3>
    <button title='Remove category' className='button button--link button--hover-close' onClick={() => buttonAction(category.id)}>
      <svg className='icon' viewBox='0 0 12 12' width='12' height='12'>
        <path fillRule='evenodd' d='M11.53.47a.75.75 0 0 0-1.061 0l-4.47 4.47L1.529.47A.75.75 0 1 0 .468 1.531l4.47 4.47-4.47 4.47a.75.75 0 1 0 1.061 1.061l4.47-4.47 4.47 4.47a.75.75 0 1 0 1.061-1.061l-4.47-4.47 4.47-4.47a.75.75 0 0 0 0-1.061z' />
      </svg>
    </button>
  </div>
)

EditCategoryItem.propTypes = {
  buttonAction: PropTypes.func.isRequired,
  category: PropTypes.object
}

export default EditCategoryItem
