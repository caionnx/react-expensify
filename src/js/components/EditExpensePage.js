import React from 'react'
import Modal from 'react-modal'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'

export class EditExpensePage extends React.Component {
  state = {
    removeExpenseModal: false
  }
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense).then(() => {
      this.props.history.push('/dashboard')
    })
  }
  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id }).then(() => {
      this.props.history.push('/dashboard')
    })
  }
  onOpenRemoveExpenseModal = () => {
    this.setState({ removeExpenseModal: true })
  }
  onCloseRemoveExpenseModal = () => {
    this.setState({ removeExpenseModal: false })
  }
  componentWillMount () {
    if (!this.props.expense || !this.props.expense.id) {
      this.props.history.push('/dashboard')
    }
  }
  render () {
    return (
      <div>
        <div className='page-header'>
          <div className='content-container'>
            <h1 className='page-header__title'>Edit Expense</h1>
          </div>
        </div>
        <div className='content-container'>
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button
            id='openRemoveExpenseModal'
            className='button button--secondary'
            onClick={this.onOpenRemoveExpenseModal}>
            Remove Expense
          </button>

          <Modal
            isOpen={this.state.removeExpenseModal}
            onRequestClose={this.onCloseRemoveExpenseModal}
            appElement={document.getElementById('app')}
            contentLabel='Remove Expense Modal'
            closeTimeoutMS={200}
            className='modal'>

            <p className='modal__body'>Are you sure you want to remove this expense?</p>
            <div className='modal__options'>
              <button id='closeRemoveExpenseModal' className='button' onClick={this.onCloseRemoveExpenseModal}>Cancel</button>
              <button id='removeExpense' className='button button--secondary' onClick={this.onRemove}>Remove</button>
            </div>
          </Modal>
        </div>
      </div>
    )
  }
}

EditExpensePage.propTypes = {
  expense: PropTypes.shape({
    id: PropTypes.string
  }),
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }),
  startEditExpense: PropTypes.func.isRequired,
  startRemoveExpense: PropTypes.func.isRequired
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)
