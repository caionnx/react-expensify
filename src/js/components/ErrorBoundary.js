import React from 'react'
import PropTypes from 'prop-types'

class ErrorBoundary extends React.Component {
  state = { error: null }

  componentDidCatch (error) {
    this.setState({ error })
  }

  render () {
    const { error } = this.state

    if (!error) {
      return this.props.children
    }

    return (
      <div className='content-container'>
        <h2>Something went wrong loading this page. Please try to reload the app.</h2>
        <details>
          <summary>Details for nerds</summary>
          {error.toString && error.toString()}
        </details>
      </div>
    )
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.any
}

export default ErrorBoundary
