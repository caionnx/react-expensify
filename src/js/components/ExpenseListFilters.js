import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import DayPickerInput from 'react-day-picker/DayPickerInput'
import dateFormat from 'date-fns/format'
import ExpensesCategorySelect from './ExpensesCategorySelect'
import {
  setTextFilter,
  sortByDate,
  sortByAmount,
  setStartDate,
  setEndDate,
  setCategory
} from '../actions/filters'

export class ExpenseListFilters extends React.Component {
  onClearDates = () => {
    this.props.setStartDate('')
    this.props.setEndDate('')
  }
  onStartDateChange = (startDate) => {
    this.props.setStartDate(startDate)
  }
  onEndDateChange = (endDate) => {
    this.props.setEndDate(endDate)
  }
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value)
  }
  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate()
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount()
    }
  }
  onCategoryChange = (category) => {
    this.props.setCategory(category)
  }
  render () {
    const {
      startDate,
      endDate,
      text,
      category,
      sortBy
    } = this.props.filters

    return (
      <div className='content-container'>
        <div className='input-group'>
          <div className='input-group__item'>
            <input
              className='text-input'
              placeholder='Search expenses'
              type='text'
              value={text}
              onChange={this.onTextChange}
            />
          </div>
          <div className='input-group__item'>
            <ExpensesCategorySelect
              onChange={this.onCategoryChange}
              defaultValue={category || ''}
              defaultText='All categories' />
          </div>
          <div className='input-group__item'>
            <select
              className='select'
              value={sortBy}
              onChange={this.onSortChange}
            >
              <option value='date'>Date</option>
              <option value='amount'>Amount</option>
            </select>
          </div>
          <div className='DayPickerContainer'>
            <DayPickerInput
              format='MM/DD/YYYY'
              value={startDate && dateFormat(startDate, 'MM/DD/YYYY')}
              placeholder='Start Date'
              dayPickerProps={{
                selectedDays: [startDate, { from: startDate, to: endDate }],
                disabledDays: { after: endDate },
                toMonth: endDate,
                modifiers: { start: startDate, end: endDate },
                numberOfMonths: 1,
                onDayClick: () => this.endDate.getInput().focus()
              }}
              onDayChange={this.onStartDateChange}
            />
            <DayPickerInput
              ref={el => (this.endDate = el)}
              format='MM/DD/YYYY'
              value={endDate && dateFormat(endDate, 'MM/DD/YYYY')}
              placeholder='End Date'
              dayPickerProps={{
                selectedDays: [startDate, { from: startDate, to: endDate }],
                disabledDays: { before: startDate },
                modifiers: { start: startDate, end: endDate },
                month: startDate,
                fromMonth: startDate,
                numberOfMonths: 1
              }}
              onDayChange={this.onEndDateChange}
            />
            <button onClick={this.onClearDates} type='button' aria-label='Clear Dates' className=''>
              <svg className='icon' viewBox='0 0 12 12' width='12' height='12'>
                { /* eslint-disable-next-line  */}
                <path fillRule='evenodd' d='M11.53.47a.75.75 0 0 0-1.061 0l-4.47 4.47L1.529.47A.75.75 0 1 0 .468 1.531l4.47 4.47-4.47 4.47a.75.75 0 1 0 1.061 1.061l4.47-4.47 4.47 4.47a.75.75 0 1 0 1.061-1.061l-4.47-4.47 4.47-4.47a.75.75 0 0 0 0-1.061z'></path>
              </svg>
            </button>
          </div>
        </div>
      </div>
    )
  }
}

ExpenseListFilters.propTypes = {
  filters: PropTypes.shape({
    endDate: PropTypes.oneOfType([ PropTypes.instanceOf(Date), PropTypes.string ]),
    sortBy: PropTypes.string,
    startDate: PropTypes.oneOfType([ PropTypes.instanceOf(Date), PropTypes.string ]),
    text: PropTypes.string,
    category: PropTypes.string
  }),
  setEndDate: PropTypes.func.isRequired,
  setStartDate: PropTypes.func.isRequired,
  setTextFilter: PropTypes.func.isRequired,
  sortByAmount: PropTypes.func.isRequired,
  sortByDate: PropTypes.func.isRequired,
  setCategory: PropTypes.func.isRequired
}

const mapStateToProps = (state) => ({
  filters: state.filters
})

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  setCategory: (category) => dispatch(setCategory(category))
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)
