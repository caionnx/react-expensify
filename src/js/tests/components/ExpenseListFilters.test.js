import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseListFilters, default as ConnectedExpenseListFilters } from '../../components/ExpenseListFilters'
import { filters, altFilters } from '../fixtures/filters'
import configureMockStore from 'redux-mock-store'
jest.mock('date-fns/format', () =>
  (date) => {
    if (date.match('1970-01-01')) return '01/01/1970'
    if (date.match('1970-01-04')) return '04/01/1970'
  }
)
const mockStore = configureMockStore()

let setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate, setCategory, wrapper

beforeEach(() => {
  setTextFilter = jest.fn()
  sortByDate = jest.fn()
  sortByAmount = jest.fn()
  setStartDate = jest.fn()
  setEndDate = jest.fn()
  setCategory = jest.fn()
  wrapper = shallow(
    <ExpenseListFilters
      filters={filters}
      setTextFilter={setTextFilter}
      sortByDate={sortByDate}
      sortByAmount={sortByAmount}
      setStartDate={setStartDate}
      setEndDate={setEndDate}
      setCategory={setCategory}
    />
  )
})

test('should render ExpenseListFilters correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseListFilters with alt data correctly', () => {
  wrapper.setProps({
    filters: altFilters
  })
  expect(wrapper).toMatchSnapshot()
})

test('should handle text change', () => {
  const value = 'rent'
  wrapper.find('input').simulate('change', {
    target: { value }
  })
  expect(setTextFilter).toHaveBeenLastCalledWith(value)
})

test('should sort by date', () => {
  const value = 'date'
  wrapper.setProps({
    filters: altFilters
  })
  wrapper.find('select').simulate('change', {
    target: { value }
  })
  expect(sortByDate).toHaveBeenCalled()
})

test('should sort by amount', () => {
  const value = 'amount'
  wrapper.find('select').simulate('change', {
    target: { value }
  })
  expect(sortByAmount).toHaveBeenCalled()
})

test('should call setCategory when component change', () => {
  wrapper.find('Connect(ExpensesCategorySelect)').prop('onChange')()
  expect(setCategory).toHaveBeenCalled()
})

test('should handle date changes', () => {
  const startDate = new Date()
  const endDate = new Date()
  wrapper.find('DayPickerInput').at(0).prop('onDayChange')({ startDate })
  wrapper.find('DayPickerInput').at(1).prop('onDayChange')({ endDate })
  expect(setStartDate).toHaveBeenLastCalledWith({ startDate })
  expect(setEndDate).toHaveBeenLastCalledWith({ endDate })
})

test('should clear dates on button click', () => {
  wrapper.find('button').simulate('click')
  expect(setStartDate).toHaveBeenLastCalledWith('')
  expect(setEndDate).toHaveBeenLastCalledWith('')
})

test('should have mapStateToProps and mapDispatchToProps', () => {
  const store = mockStore({ filters })

  wrapper = shallow(<ConnectedExpenseListFilters store={store} />)

  expect(wrapper.prop('filters')).toEqual(filters)
  expect(wrapper.prop('setTextFilter')()).toMatchObject({ type: expect.any(String) }) // Expect mapDispatch to generate an action object
  expect(wrapper.prop('sortByDate')()).toMatchObject({ type: expect.any(String) })
  expect(wrapper.prop('sortByAmount')()).toMatchObject({ type: expect.any(String) })
  expect(wrapper.prop('setStartDate')()).toMatchObject({ type: expect.any(String) })
  expect(wrapper.prop('setEndDate')()).toMatchObject({ type: expect.any(String) })
  expect(wrapper.prop('setCategory')()).toMatchObject({ type: expect.any(String) })
})
