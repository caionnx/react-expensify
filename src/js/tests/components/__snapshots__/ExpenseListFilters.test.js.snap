// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListFilters correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search expenses"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group__item"
    >
      <Connect(ExpensesCategorySelect)
        defaultText="All categories"
        defaultValue=""
        onChange={[Function]}
      />
    </div>
    <div
      className="input-group__item"
    >
      <select
        className="select"
        onChange={[Function]}
        value="date"
      >
        <option
          value="date"
        >
          Date
        </option>
        <option
          value="amount"
        >
          Amount
        </option>
      </select>
    </div>
    <div
      className="DayPickerContainer"
    >
      <DayPickerInput
        classNames={
          Object {
            "container": "DayPickerInput",
            "overlay": "DayPickerInput-Overlay",
            "overlayWrapper": "DayPickerInput-OverlayWrapper",
          }
        }
        clickUnselectsDay={false}
        component="input"
        dayPickerProps={
          Object {
            "disabledDays": Object {
              "after": undefined,
            },
            "modifiers": Object {
              "end": undefined,
              "start": undefined,
            },
            "numberOfMonths": 1,
            "onDayClick": [Function],
            "selectedDays": Array [
              undefined,
              Object {
                "from": undefined,
                "to": undefined,
              },
            ],
            "toMonth": undefined,
          }
        }
        format="L"
        formatDate={[Function]}
        hideOnDayClick={true}
        inputProps={Object {}}
        keepFocus={true}
        onDayChange={[Function]}
        overlayComponent={[Function]}
        parseDate={[Function]}
        placeholder="Start Date"
        showOverlay={false}
        value=""
      />
      <DayPickerInput
        classNames={
          Object {
            "container": "DayPickerInput",
            "overlay": "DayPickerInput-Overlay",
            "overlayWrapper": "DayPickerInput-OverlayWrapper",
          }
        }
        clickUnselectsDay={false}
        component="input"
        dayPickerProps={
          Object {
            "disabledDays": Object {
              "before": undefined,
            },
            "fromMonth": undefined,
            "modifiers": Object {
              "end": undefined,
              "start": undefined,
            },
            "month": undefined,
            "numberOfMonths": 1,
            "selectedDays": Array [
              undefined,
              Object {
                "from": undefined,
                "to": undefined,
              },
            ],
          }
        }
        format="L"
        formatDate={[Function]}
        hideOnDayClick={true}
        inputProps={Object {}}
        keepFocus={true}
        onDayChange={[Function]}
        overlayComponent={[Function]}
        parseDate={[Function]}
        placeholder="End Date"
        showOverlay={false}
        value=""
      />
      <button
        aria-label="Clear Dates"
        className=""
        onClick={[Function]}
        type="button"
      >
        <svg
          className="icon"
          height="12"
          viewBox="0 0 12 12"
          width="12"
        >
          <path
            d="M11.53.47a.75.75 0 0 0-1.061 0l-4.47 4.47L1.529.47A.75.75 0 1 0 .468 1.531l4.47 4.47-4.47 4.47a.75.75 0 1 0 1.061 1.061l4.47-4.47 4.47 4.47a.75.75 0 1 0 1.061-1.061l-4.47-4.47 4.47-4.47a.75.75 0 0 0 0-1.061z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseListFilters with alt data correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search expenses"
        type="text"
        value="bills"
      />
    </div>
    <div
      className="input-group__item"
    >
      <Connect(ExpensesCategorySelect)
        defaultText="All categories"
        defaultValue=""
        onChange={[Function]}
      />
    </div>
    <div
      className="input-group__item"
    >
      <select
        className="select"
        onChange={[Function]}
        value="amount"
      >
        <option
          value="date"
        >
          Date
        </option>
        <option
          value="amount"
        >
          Amount
        </option>
      </select>
    </div>
    <div
      className="DayPickerContainer"
    >
      <DayPickerInput
        classNames={
          Object {
            "container": "DayPickerInput",
            "overlay": "DayPickerInput-Overlay",
            "overlayWrapper": "DayPickerInput-OverlayWrapper",
          }
        }
        clickUnselectsDay={false}
        component="input"
        dayPickerProps={
          Object {
            "disabledDays": Object {
              "after": "1970-01-04T00:00:00.000Z",
            },
            "modifiers": Object {
              "end": "1970-01-04T00:00:00.000Z",
              "start": "1970-01-01T00:00:00.000Z",
            },
            "numberOfMonths": 1,
            "onDayClick": [Function],
            "selectedDays": Array [
              "1970-01-01T00:00:00.000Z",
              Object {
                "from": "1970-01-01T00:00:00.000Z",
                "to": "1970-01-04T00:00:00.000Z",
              },
            ],
            "toMonth": "1970-01-04T00:00:00.000Z",
          }
        }
        format="L"
        formatDate={[Function]}
        hideOnDayClick={true}
        inputProps={Object {}}
        keepFocus={true}
        onDayChange={[Function]}
        overlayComponent={[Function]}
        parseDate={[Function]}
        placeholder="Start Date"
        showOverlay={false}
        value="1970-01-01T00:00:00.000Z"
      />
      <DayPickerInput
        classNames={
          Object {
            "container": "DayPickerInput",
            "overlay": "DayPickerInput-Overlay",
            "overlayWrapper": "DayPickerInput-OverlayWrapper",
          }
        }
        clickUnselectsDay={false}
        component="input"
        dayPickerProps={
          Object {
            "disabledDays": Object {
              "before": "1970-01-01T00:00:00.000Z",
            },
            "fromMonth": "1970-01-01T00:00:00.000Z",
            "modifiers": Object {
              "end": "1970-01-04T00:00:00.000Z",
              "start": "1970-01-01T00:00:00.000Z",
            },
            "month": "1970-01-01T00:00:00.000Z",
            "numberOfMonths": 1,
            "selectedDays": Array [
              "1970-01-01T00:00:00.000Z",
              Object {
                "from": "1970-01-01T00:00:00.000Z",
                "to": "1970-01-04T00:00:00.000Z",
              },
            ],
          }
        }
        format="L"
        formatDate={[Function]}
        hideOnDayClick={true}
        inputProps={Object {}}
        keepFocus={true}
        onDayChange={[Function]}
        overlayComponent={[Function]}
        parseDate={[Function]}
        placeholder="End Date"
        showOverlay={false}
        value="1970-01-04T00:00:00.000Z"
      />
      <button
        aria-label="Clear Dates"
        className=""
        onClick={[Function]}
        type="button"
      >
        <svg
          className="icon"
          height="12"
          viewBox="0 0 12 12"
          width="12"
        >
          <path
            d="M11.53.47a.75.75 0 0 0-1.061 0l-4.47 4.47L1.529.47A.75.75 0 1 0 .468 1.531l4.47 4.47-4.47 4.47a.75.75 0 1 0 1.061 1.061l4.47-4.47 4.47 4.47a.75.75 0 1 0 1.061-1.061l-4.47-4.47 4.47-4.47a.75.75 0 0 0 0-1.061z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
